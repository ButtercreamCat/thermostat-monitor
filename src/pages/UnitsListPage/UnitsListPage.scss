.unitsList {
  width: 90%;
  max-width: 900px;
  margin: auto;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  &_title {
    font-size: 3rem;
    text-align: center;
    margin: 40px 0;
  }
}
