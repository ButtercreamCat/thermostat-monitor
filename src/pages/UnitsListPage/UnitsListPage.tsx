import React from "react";
import { Link } from "react-router-dom";
import "./UnitsListPage.scss";

const unitLists = [
  100,
  110,
  120,
  130,
  140,
  150,
  160,
  170,
  180,
  190,
  200,
  210,
  220,
  230,
  240,
  250,
  260,
  270,
  280,
  290,
  300,
  310,
  320,
  330,
  340,
  350,
  360,
  370,
  380,
  390,
  400,
  410,
  420,
  430,
  440,
  450,
  460,
  470,
  480,
  490,
  500,
  510,
  520,
  530,
  540,
  550,
  560,
  570,
  580,
  590,
  600,
  610,
  620,
  630,
  640,
  650,
  660,
  670,
  680,
  690
];

function UnitsListPage() {
  return (
    <>
      <h1 className="unitsList_title">Abc Condo</h1>
      <div className="unitsList">
        {unitLists.map(unitNumber => (
          <Link
            key={unitNumber}
            to={`/unit/${unitNumber}/thermostat`}
            className="button button-square-withIcon"
          >
            <span className="icon-house"></span>
            <div>Unit {unitNumber}</div>
          </Link>
        ))}
      </div>
    </>
  );
}

export default UnitsListPage;
