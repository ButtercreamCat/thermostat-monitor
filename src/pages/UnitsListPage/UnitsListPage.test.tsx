import React from "react";
import { shallow } from "enzyme";
import UnitsListPage from "./UnitsListPage";

test("should render UnitsListPage correctly", () => {
  const wrapper = shallow(<UnitsListPage></UnitsListPage>);
  expect(wrapper).toMatchSnapshot();
});
