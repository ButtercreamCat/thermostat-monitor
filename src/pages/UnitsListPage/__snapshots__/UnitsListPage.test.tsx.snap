// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UnitsListPage correctly 1`] = `
<Fragment>
  <h1
    className="unitsList_title"
  >
    Abc Condo
  </h1>
  <div
    className="unitsList"
  >
    <Link
      className="button button-square-withIcon"
      key="100"
      to="/unit/100/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        100
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="110"
      to="/unit/110/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        110
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="120"
      to="/unit/120/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        120
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="130"
      to="/unit/130/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        130
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="140"
      to="/unit/140/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        140
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="150"
      to="/unit/150/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        150
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="160"
      to="/unit/160/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        160
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="170"
      to="/unit/170/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        170
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="180"
      to="/unit/180/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        180
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="190"
      to="/unit/190/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        190
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="200"
      to="/unit/200/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        200
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="210"
      to="/unit/210/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        210
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="220"
      to="/unit/220/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        220
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="230"
      to="/unit/230/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        230
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="240"
      to="/unit/240/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        240
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="250"
      to="/unit/250/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        250
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="260"
      to="/unit/260/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        260
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="270"
      to="/unit/270/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        270
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="280"
      to="/unit/280/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        280
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="290"
      to="/unit/290/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        290
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="300"
      to="/unit/300/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        300
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="310"
      to="/unit/310/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        310
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="320"
      to="/unit/320/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        320
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="330"
      to="/unit/330/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        330
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="340"
      to="/unit/340/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        340
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="350"
      to="/unit/350/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        350
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="360"
      to="/unit/360/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        360
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="370"
      to="/unit/370/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        370
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="380"
      to="/unit/380/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        380
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="390"
      to="/unit/390/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        390
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="400"
      to="/unit/400/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        400
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="410"
      to="/unit/410/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        410
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="420"
      to="/unit/420/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        420
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="430"
      to="/unit/430/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        430
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="440"
      to="/unit/440/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        440
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="450"
      to="/unit/450/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        450
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="460"
      to="/unit/460/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        460
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="470"
      to="/unit/470/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        470
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="480"
      to="/unit/480/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        480
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="490"
      to="/unit/490/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        490
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="500"
      to="/unit/500/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        500
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="510"
      to="/unit/510/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        510
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="520"
      to="/unit/520/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        520
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="530"
      to="/unit/530/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        530
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="540"
      to="/unit/540/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        540
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="550"
      to="/unit/550/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        550
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="560"
      to="/unit/560/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        560
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="570"
      to="/unit/570/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        570
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="580"
      to="/unit/580/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        580
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="590"
      to="/unit/590/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        590
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="600"
      to="/unit/600/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        600
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="610"
      to="/unit/610/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        610
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="620"
      to="/unit/620/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        620
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="630"
      to="/unit/630/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        630
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="640"
      to="/unit/640/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        640
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="650"
      to="/unit/650/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        650
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="660"
      to="/unit/660/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        660
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="670"
      to="/unit/670/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        670
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="680"
      to="/unit/680/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        680
      </div>
    </Link>
    <Link
      className="button button-square-withIcon"
      key="690"
      to="/unit/690/thermostat"
    >
      <span
        className="icon-house"
      />
      <div>
        Unit 
        690
      </div>
    </Link>
  </div>
</Fragment>
`;
